### PR TITLE
Support for suspend function generation

### PR DESCRIPTION
**Pull request by minaEweida**

### CODE REVIEW

File: samples/sample/src/commonMain/kotlin/com/careem/mockingbird/sample/PippoSample.kt [hunk 1]
```diff
@@ -24,4 +24,5 @@ interface PippoSample {
     fun sayHiWith(param: String)
     fun sayHiWith(param: String, someOtherParam: Boolean)
     fun sayHiWithCommonParam(param: String, intParam: Int)
+    suspend fun thisIsSuspend(param: String, intParam: Int)
```
Comment (carlonzo) [inline samples/sample/src/commonMain/kotlin/com/careem/mockingbird/sample/PippoSample.kt:27]: I dont see u have any tests for this

**Review by carlonzo — APPROVED** [review]
approved but I would suggest to add this case in the tests

**Review by MarcoSignoretto — APPROVED** [review]

File: samples/sample/src/commonMain/kotlin/com/careem/mockingbird/sample/PippoSample.kt (hunk 1 above)
Comment (minaEweida) [reply samples/sample/src/commonMain/kotlin/com/careem/mockingbird/sample/PippoSample.kt:27]: we are going to write a test for code generation and will include this one as well

Comment (minaEweida) [reply samples/sample/src/commonMain/kotlin/com/careem/mockingbird/sample/PippoSample.kt:27]: #95

**Pull request merged**